stdlib/buffer.ml

```diff
@@ -350,6 +350,15 @@ let truncate b len =
     else
       b.position <- len
 
+let drop b len =
+    if len < 0 || len > b.position then
+      invalid_arg "Buffer.drop"
```

Comment:
Also it is customary for `drop` functions not to error if you try to drop too much. See e.g. Haskell's `drop`, Batteries `{l,r}chop`, Containers `drop`. Base `drop_{prefix,suffix}`.